server/src/main/java/io/crate/metadata/doc/DocTableInfoFactory.java

```diff
@@ -556,8 +556,11 @@ private static IndexType getColumnIndexType(Map<String, Object> columnProperties
      * Array types have the mapping for their inner type within `inner`
      **/
     private static Map<String, Object> innerProperties(Map<String, Object> columnProperties) {
-        Map<String, Object> inner = Maps.get(columnProperties, "inner");
-        return inner == null ? columnProperties : inner;
+        Map<String, Object> inner = columnProperties;
+        while (inner.containsKey("inner")) {
```

Comment:
minor, but maybe use a `while (inner != null)` and a helper var to avoid `contains` in the loop, since `get` is already called inside the loop.